src/browser/api_protocol/api_handlers/application.js

```diff
@@ -341,14 +342,14 @@ function ApplicationApiHandler() {
     function deregisterExternalWindow(identity, message, ack) {
         const windowIdentity = apiProtocolBase.getTargetWindowIdentity(message.payload);
 
-        ofEvents.emit(`external-window/close/${windowIdentity.uuid}-${windowIdentity.name}`);
+        ofEvents.emit(route.externalWindow('close', windowIdentity.uuid, windowIdentity.name));
         ack(successAck);
     }
 
     function externalWindowAction(identity, message, ack) {
         /* jshint bitwise: false */
         let payload = message.payload;
-        let uuidname = `${payload.uuid}-${payload.name}`;
+        const emitToExtWin = (type, data) => ofEvents.emit(route.externalWindow(type, payload.uuid, payload.name), data);
```

Comment:
Removed to a separate PR:
* `emitToExtWin` has **11** calls in src/browser/api_protocol/api_handlers/application.js
* `emitToApp` has **8** calls in src/browser/api/application.js (similar; calls `route.application` _and_ `ofEvents.emit`):
* `routeExtWin` has **10** calls in src/browser/external_window_event_adapter.js (simpler; just calls `route.window`)